tests/tck/utils/comparator.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) 2020 vesoft inc. All rights reserved.
+#
+# This source code is licensed under Apache 2.0 License,
+# attached with Common Clause Condition 1.0, found in the LICENSES directory.
+
+import math
+
+from nebula2.data.DataObject import (
+    Node,
+    Record,
+    Relationship,
+    Path,
+    DataSetWrapper,
+    ValueWrapper,
+)
+
+
+class DataSetWrapperComparator:
+    def __init__(self, strict=True, order=False):
+        self._strict = strict
+        self._order = order
+
+    def __call__(self, lhs: DataSetWrapper, rhs: DataSetWrapper):
+        return self.compare(lhs, rhs)
+
+    def compare(self, lhs: DataSetWrapper, rhs: DataSetWrapper):
+        if lhs.get_row_size() != rhs.get_row_size():
+            return False
+        if not lhs.get_col_names() == rhs.get_col_names():
+            return False
+        if self._order:
+            return all(self.compare_row(l, r) for (l, r) in zip(lhs, rhs))
+        return self._compare_list(lhs, rhs, self.compare_row)
+
+    def compare_value(self, lhs: ValueWrapper, rhs: ValueWrapper):
+        '''
+        lhs and rhs represent response data and expected data respectively
+        '''
+        if lhs.is_null():
+            return rhs.is_null()
```

Comment:
Shall we elaborate the individual null types?

Reply:
Maybe. But we can let it go at present. When we encounter those test cases, it's not late to fix it. Will add some todo comments.